log4j-core/src/main/java/org/apache/logging/log4j/core/layout/Rfc5424Layout.java

```diff
@@ -92,11 +97,12 @@
      */
     public static final String DEFAULT_MDCID = "mdc";
 
+    private static final FixedDateFormat FIXED_DATE_FORMAT_MILLI_PRECISION = FixedDateFormat
+            .create(FixedFormat.ISO8601_OFFSET_DATE_TIME_HHCMM_PERIOD);
+    private static final FixedDateFormat FIXED_DATE_FORMAT_MICRO_PRECISION = FixedDateFormat
+            .create(FixedFormat.ISO8601_OFFSET_DATE_TIME_HHCMM_PERIOD_MICRO);
+
```

Comment:
I would use a (non static) `org.apache.logging.log4j.core.pattern.DatePatternConverter` to keep the layout garbage free.